spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/annotation/LoadBalancerClientConfiguration.java

```diff
@@ -109,23 +86,6 @@ public ServiceInstanceListSupplier discoveryClientServiceInstanceListSupplier(
 					.withCaching().build(context);
 		}
 
-		@Bean
```

Comment:
I think it's just to replace the deprecated class, not to delete it completely

Reply:
The class is already deprecated in 2.2.x and can be removed in master.

Reply:
But we still need serviceinstancelistsupplier bean. We can modify it in this way:
@Bean
		@ConditionalOnBean(ReactiveDiscoveryClient.class)
		@ConditionalOnMissingBean
		public ServiceInstanceListSupplier discoveryClientServiceInstanceSupplier(
				ReactiveDiscoveryClient discoveryClient, Environment env,
				ApplicationContext context) {
			DiscoveryClientServiceInstanceListSupplier delegate = new DiscoveryClientServiceInstanceListSupplier(
					discoveryClient, env);
			ObjectProvider<LoadBalancerCacheManager> cacheManagerProvider = context
					.getBeanProvider(LoadBalancerCacheManager.class);
			if (cacheManagerProvider.getIfAvailable() != null) {
				return new CachingServiceInstanceListSupplier(delegate,
						cacheManagerProvider.getIfAvailable());
			}
			return delegate;
		}

Reply:
That is already in this class